testsuite/classlibrary/TestReadableNodeIDAllocator.sc

```diff
@@ -0,0 +1,85 @@
+TestReadableNodeIDAllocator : UnitTest {
+
+	// test proper creation from Server
+	test_readableNodeRange {
+		var s = Server(\testRNIA);
+		var prevClass = Server.nodeAllocClass;
+		Server.nodeAllocClass = ReadableNodeIDAllocator;
+		s.options.maxLogins = 1; s.newAllocators;
+		// this.assert(
+		// 	(s.nodeAllocator.numIDs == (10 ** 8)),
+		// 	"for a single client, (readable) nodeAllocator has maximum range."
+		// );
+		// this.assert(
+		// 	(s.nodeAllocator.numIDs == (10 ** 7)),
+		// 	"for 16 clients, (readable) nodeAllocator has divided range."
+		// );
+
+		// same for ReadableNodeIDAllocator
+		// Server.nodeAllocClass = ReadableNodeIDAllocator;
+		Server.named.removeAt(s.name); Server.all.remove(s);
+		Server.nodeAllocClass = prevClass;
+	}
+
+	test_permIDs {
+		var alloc = ReadableNodeIDAllocator(8191, 1000, 8191);
+		var nextPermID, permIDs = List[], permIDToFree;
+
+		this.assert(
+			alloc.isPerm(alloc.idOffset - 1).not
+			and: alloc.isPerm(alloc.idOffset)
+			and: alloc.isPerm(alloc.idOffset + alloc.lowestTempID - 1)
+			and: alloc.isPerm(alloc.idOffset + alloc.lowestTempID).not,
+			"ReadableNodeIDAllocator should identify its permanent IDs."
+		);
+		while {
+			if (0.5.coin and: { permIDs.size > 0 }) {
+				permIDToFree = permIDs.remove(permIDs.choose);
+				alloc.freePerm(permIDToFree);
+			};
+			nextPermID = alloc.allocPerm;
+			nextPermID.notNil;
+		} {
+			permIDs.add(nextPermID);
+		};
+		this.assert(
+			permIDs.size.postln == (alloc.lowestTempID - 2),
+			"ReadableNodeIDAllocator should hand out all permanent IDs in mixed alloc/free use."
+		);
+
+		while {
+			if (0.5.coin) {
```

Comment:
please avoid randomness in tests

Reply:
randSeed still operating here.

Reply:
??

Reply:
In the beginning of `test_permIDs`, the random seed of the current thread is set:

```
thisThread.randSeed_(4711);
```

This will cause all subsequent random decisions within that thread to be deterministic: each time you call `permIDs`, they are the same.

Reply:
got it, thanks!